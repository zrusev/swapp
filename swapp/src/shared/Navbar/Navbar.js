import React from 'react';
import { func } from 'prop-types';
import { Link } from 'react-router-dom';
import {
    Header,
    Container,
    Logo,
    Wrapper,
    Item,
    ExitContainer,
    ExitButton,
} from './components/_index';

const Navbar = ({ toggleTheme }) => {
    return (
        <Header>
            <Container>
                <Logo onClick={toggleTheme}>
                    <span href='#'>
                        <h1>SWAPP</h1>
                    </span>
                </Logo>
                <Wrapper>
                    <Item>
                        <Link to='/episodes'>
                            Episodes
                        </Link>
                    </Item>
                    <Item>
                        <Link to='/characters'>
                            Characters
                        </Link>
                    </Item>
                    <ExitContainer onClick={() => {}}>
                        <span>
                            <ExitButton />
                        </span>
                    </ExitContainer>
                </Wrapper>
            </Container>
        </Header>
    )
}

Navbar.propTypes = {
    toggleTheme: func.isRequired,
}

export default Navbar;