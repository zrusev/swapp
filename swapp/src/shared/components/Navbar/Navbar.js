import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from 'styled-components';
import useNavbar from './hooks/useNavbar';
import { CSSTransition } from 'react-transition-group';
import HamburgerMenu from 'react-hamburger-menu';

import ThemeToggle from '../ThemeToggle/ThemeToggle';
import ExitButton from '../ExitButton/ExitButton';

import {
  Header,
  Wrapper,
  Hamburger,
  NavAnimation,
  Link,
} from './components/_index';

const Navbar = ({
  theme: { primaryHeadingFontColor },
  logoutHandler,
  children,
}) => {
  const { isNavVisible, isSmallScreen, toggleNav } = useNavbar();

  return (
    <>
      <Header>
        <ThemeToggle>
          <span>SWAPP</span>
        </ThemeToggle>
        <CSSTransition
          in={!isSmallScreen || isNavVisible}
          timeout={350}
          classNames="NavAnimation"
          unmountOnExit
        >
          {() => (
            <NavAnimation>
              <Wrapper>
                <Link to="/episodes">Episodes</Link>
                <Link to="/characters">Characters</Link>
                <ExitButton action={logoutHandler} />
              </Wrapper>
            </NavAnimation>
          )}
        </CSSTransition>
        <Hamburger onClick={toggleNav}>
          <HamburgerMenu
            isOpen={isNavVisible}
            menuClicked={() => {}}
            width={18}
            height={15}
            strokeWidth={2}
            rotate={0}
            color={primaryHeadingFontColor}
            borderRadius={0}
            animationDuration={0.5}
          />
        </Hamburger>
      </Header>
      <main>{children}</main>
    </>
  );
};

Navbar.propTypes = {
  theme: PropTypes.object.isRequired,
  children: PropTypes.array.isRequired,
  logoutHandler: PropTypes.func.isRequired,
};

export default withTheme(Navbar);
