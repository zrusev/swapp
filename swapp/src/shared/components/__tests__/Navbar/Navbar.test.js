import React from 'react';
import { shallow } from 'enzyme';

import Navbar from '../../Navbar/Navbar';

describe('should Navbar be styled', () => {
  const props = {
    theme: { primaryHeadingFontColor: '' },
    logoutHandler: jest.fn(),
  };

  it('should match description', () => {
    const wrapper = shallow(
      <Navbar {...props}>
        <span>Navigation</span>
      </Navbar>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
