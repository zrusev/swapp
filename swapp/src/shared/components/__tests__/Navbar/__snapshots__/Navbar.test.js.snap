// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Navbar be styled should match description 1`] = `
<Navbar
  logoutHandler={[MockFunction]}
  theme={
    Object {
      "primaryHeadingFontColor": "",
    }
  }
>
  <span>
    Navigation
  </span>
</Navbar>
`;
